features/dd-sdk-android-trace/src/main/java/com/datadog/opentracing/PendingTrace.java

```diff
@@ -84,9 +89,27 @@ public long getCurrentTimeNano() {
 
   public void registerSpan(final DDSpan span) {
     if (traceId == null || span.context() == null) {
+      internalLogger.log(
+        InternalLogger.Level.ERROR,
+        InternalLogger.Target.USER,
+        () -> "Span " + span.getOperationName() + " not registered because of null traceId or context; "+
+          "spanId:" + span.getSpanId() + " traceid:" + traceId,
+        null,
+        false,
+        new HashMap<>()
+      );
       return;
     }
     if (!traceId.equals(span.context().getTraceId())) {
+      internalLogger.log(
+        InternalLogger.Level.ERROR,
+        InternalLogger.Target.USER,
+        () -> "Span " + span.getOperationName() + " not registered because of traceId mismatch; " +
```

Comment:
is it useful here also to log the value of `span.context().getTraceId()`?

Reply:
Makes sense, yes.